app/uk/gov/hmrc/apisubscriptionfields/service/FieldsDefinitionService.scala

```diff
@@ -29,22 +29,9 @@ import scala.concurrent.Future
 class FieldsDefinitionService @Inject() (repository: FieldsDefinitionRepository) {
 
   def upsert(identifier: FieldsDefinitionIdentifier, fields: Seq[FieldDefinition]): Future[Boolean] = {
-    def update(existingFieldsDefinitionId: String): Future[Boolean] =
-      save(FieldsDefinition(existingFieldsDefinitionId, fields))
-
-    def create(): Future[Boolean] =
-      save(FieldsDefinition(identifier.encode(), fields))
-
     Logger.debug(s"[upsert] FieldsDefinitionIdentifier: $identifier")
 
-    repository.fetchById(identifier.encode()) flatMap {
-      o =>
-        o.fold(
-          create() map { _ => true }
-        )(
-          existing => update(existing.id) map { _ => false }
-        )
-    }
+    repository.upsert(FieldsDefinition(identifier.encode(), fields)) map (isInserted => isInserted)
```

Comment:
What is the purpose of adding `map ( x => x )` here?
Does it do anything useful?
For me it seems useless.
You are mapping something to itself, without any transformation.

Reply:
done